include/swift/AST/DiagnosticsParse.def

```diff
@@ -686,6 +686,18 @@ ERROR(sil_witness_assoc_conf_not_found,none,
 ERROR(sil_witness_protocol_conformance_not_found,none,
       "sil protocol conformance not found", ())
 
+// SIL differentiability witnesses
+ERROR(sil_diff_witness_expected_keyword,PointsToFirstBadToken,
+      "expected '%0' in differentiability witness", (StringRef))
+ERROR(sil_diff_witness_expected_parameter_list,PointsToFirstBadToken,
+      "expected an comma-separated list of parameter indices, e.g. (0, 1)", ())
+ERROR(sil_diff_witness_expected_rsquare,PointsToFirstBadToken,
+      "expected ']' to end 'differentiable' attribute", ())
+ERROR(sil_diff_witness_expected_parameter_index,PointsToFirstBadToken,
+      "expected the index of a parameter to differentiate w.r.t.", ())
+ERROR(sil_diff_witness_expected_source_index,PointsToFirstBadToken,
```

Comment:
The word "source" has been a bad word in the implementation for a long time ([SR-8888](https://bugs.swift.org/browse/SR-8888)) -- please avoid using it. "Differentiate from" is also bad. We should use "result index" and "differentiate with respect to".

Reply:
Copied code, sorry. Updated in https://github.com/apple/swift/pull/27487/commits/771143ae652d5b9cc49fc152863b4d133ee9c9c2.